src/study.js

```diff
@@ -0,0 +1,57 @@
+import { Rally, runStates } from "@mozilla/rally";
+import { onPageData, stopMeasurement } from "./attention-reporter";
+  
+function collectEventDataAndSubmit(rally, devMode) {
+  // note: onPageData calls startMeasurement.
+  onPageData.addListener(async (data) => {
+    if (devMode) {
+      console.debug('RS01.event', data);
+    }
+    // though we collect the data as two different event types using Web Science,
+    // we send the payload using one schema, "RS01.event".
+    // Once https://github.com/mozilla-rally/web-science/issues/33 is resolved,
+    // we will change the collection schema (but keep this pipeline schema the same).
+    rally.sendPing("RS01.event", data);
+  }, {
+      matchPatterns: ["<all_urls>"],
+      privateWindows: false
+  });
+}
+
+export default async function runStudy(devMode) {
+    const rally = new Rally();
+
+    await rally.initialize(
+        // A sample key id used for encrypting data.
+        "sample-invalid-key-id",
+        // A sample *valid* JWK object for the encryption.
+    {
+        "kty":"EC",
+        "crv":"P-256",
+        "x":"f83OJ3D2xF1Bg8vub9tLe1gHMzV76e8Tus9uPHvRVEU",
+        "y":"x_FEzRu9m36HLN_tue659LNpXW6pCyStikYjKIWI5a0",
+        "kid":"Public key used in JWS spec Appendix A.3 example"
+    },
+    // The following constant is automatically provided by
+    // the build system.
+    devMode,
+    (newState) => {
```

Comment:
I know this is from the study template, but I wonder if we could do something more readable with an async function. Mixing promises and async functions always gets a little strange to read :)

Reply:
If we do anything it'd be to file an issue on the study template, not important now though.

Reply:
yeah I think we should fix this in the study template. I went ahead and fixed this here anyway since it was a pretty easy thing. fixed here https://github.com/mozilla-rally/rally-study-01/pull/73/files#diff-96cfb484ce53e646e6de5e721925abb32111a470bbc703d44ede627e21c8a802R23